src/django_components/component_registry.py

```diff
@@ -462,12 +463,39 @@ def _register_to_library(
         component: Type["Component"],
     ) -> ComponentRegistryEntry:
         # Lazily import to avoid circular dependencies
-        from django_components.templatetags.component_tags import component as do_component
-
-        formatter = get_tag_formatter(self)
-        tag = register_tag_from_formatter(self, do_component, formatter, comp_name)
-
-        return ComponentRegistryEntry(cls=component, tag=tag)
+        from django_components.component import ComponentNode
+
+        registry = self
+
+        # Define a tag function that pre-processes the tokens, extracting
+        # the component name and passing the rest to the actual tag function.
+        def tag_fn(parser: Parser, token: Token) -> ComponentNode:
+            # Let the TagFormatter pre-process the tokens
+            bits = token.split_contents()
```

Comment:
This couple of next lines previously lived in `templatetags/component_tags.py`  (https://github.com/EmilStenstrom/django-components/pull/910/files#diff-7345171ed38e1dd2e4b8199b65a2fd082323bdb331ff0ef0dc0dd3bd43f111aeL498).

It calls the TagFormatter to figure out which start and end template tags should be used with a particular component (e.g. `{% component "table" %} ... {% endcomponent %}` vs `{% table %} ... {% endtable %}`.

I moved it up here, so that, instead, the `ComponentNode` only accepts the already-resolved `start_tag` and `end_tag`.